hawkular-command-gateway-war/src/main/java/org/hawkular/cmdgw/command/bus/BusCommands.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2015 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.hawkular.cmdgw.command.bus;
+
+import javax.enterprise.context.ApplicationScoped;
+
+import org.hawkular.bus.common.BasicMessage;
+import org.hawkular.cmdgw.api.UiSessionDestination;
+import org.hawkular.cmdgw.command.NoCommandForMessageException;
+
+/**
+ * A registry of {@link BusCommands} that can operate on messages coming over a bus queue.
+ *
+ * @author <a href="https://github.com/ppalaga">Peter Palaga</a>
+ */
+@ApplicationScoped
+public class BusCommands {
+
+    private final UiSessionDestinationBusCommand uiSessionDestinationBusCommand = new UiSessionDestinationBusCommand();
+
+    /**
+     * Returns a {@link BusCommand} that should handle the given {@code requestClass}.
+     *
+     * @param requestClass the type of a request for which a processing {@link BusCommand} should be found by this
+     *        method
+     * @return a {@link BusCommand}, never {@code null}
+     * @throws NoCommandForMessageException if no {@link BusCommand} was found
+     */
+    @SuppressWarnings("unchecked")
+    public <REQ extends BasicMessage, RESP extends BasicMessage> BusCommand<REQ> getCommand(Class<REQ> requestClass)
+            throws NoCommandForMessageException {
+        if (UiSessionDestination.class.isAssignableFrom(requestClass)) {
+            return (BusCommand<REQ>) uiSessionDestinationBusCommand;
+        }
+        throw new NoCommandForMessageException("No command found for requestClass [" + requestClass.getName() + "]");
```

Comment:
add a "//TODO" comment in here, or something to denote "// If any new types of commands are added, add them to the if-else statement here". Just so we have a placeholder that will remind developers this is a place where code needs to be manually added. Otherwise, it will be hard for someone to remember this is a place that has to be touched if new commands are added later on in the future.